velox/exec/Task.cpp

```diff
@@ -2439,14 +2439,17 @@ uint64_t Task::MemoryReclaimer::reclaim(
   return memory::MemoryReclaimer::reclaim(pool, targetBytes);
 }
 
-void Task::MemoryReclaimer::abort(memory::MemoryPool* pool) {
+void Task::MemoryReclaimer::abort(
+    memory::MemoryPool* pool,
+    std::exception_ptr error) {
   auto task = ensureTask();
   if (FOLLY_UNLIKELY(task == nullptr)) {
     return;
   }
   VELOX_CHECK_EQ(task->pool()->name(), pool->name());
-  task->requestAbort().wait();
-  memory::MemoryReclaimer::abort(pool);
+  task->setError(error);
+  task->terminate(TaskState::kFailed).wait();
```

Comment:
Call taskCompletionFuture and make a change to taskCompletionFuture from
```
  if (not isRunningLocked()) {
    return ContinueFuture();
  }
```
to
```
    if (not isRunningLocked()) {
      return makeFinishFutureLocked("Task::taskCompletionFuture");
    }
```

Reply:
Let's sync off-line. I don't quite understand why we want to call taskCompletionFuture upon failure/abortion.